### ISSUE TITLE
Unable to access psql terminal

### ISSUE THREAD

**Issue by blackyzero** [vitabaks/postgresql_cluster#168]
Hello,

I am getting following error when trying to access psql terminal. This happens on all nodes
Is it cause of using different default username in vars/main.yml file

```
patroni_superuser_username: "patroni_usr"
```

```
root@pgsql01:~# sudo -u postgres psql
could not change directory to "/root": Permission denied
psql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "postgres" does not exist
root@pgsql01:~#
```

```
root@pgsql01:~# ls -alh /var/run/postgresql/.s.PGSQL.5432
srwxrwxrwx 1 postgres postgres 0 Apr 25 14:26 /var/run/postgresql/.s.PGSQL.5432
root@pgsql01:~#

```

Thank you.

**Comment by vitabaks** [comment]
Hi @blackyzero 

Please connect to the database and show the list of users

`psql -U patroni_usr -d postgres`

`\du`

**Comment by vitabaks** [comment]
`patroni_superuser_username` it name for the superuser, set during initialization (initdb) 

https://patroni.readthedocs.io/en/latest/SETTINGS.html#postgresql 

I guess patroni doesn't create a postgres user by default

**Comment by blackyzero** [comment]
Thanks for quick respond.
It gives me following

```
root@pgnode01:/etc/postgresql/14/main# psql -U patroni_usr -d postgres
psql (14.2 (Debian 14.2-1.pgdg110+1))
Type "help" for help.

postgres=# \du
                                          List of roles
       Role name        |                         Attributes                         | Member of
------------------------+------------------------------------------------------------+-----------
 patroni_replicator_usr | Replication                                                | {}
 patroni_usr            | Superuser, Create role, Create DB, Replication, Bypass RLS | {}

postgres=#
```
How could i correct it?

**Comment by vitabaks** [comment]
Specify the user in `vars/main.yml`

```yaml
postgresql_users:
  - {name: "postgres", password: "postgres-pass", flags: "SUPERUSER"}
```

Execute playbook 
`ansible-playbook deploy_pgcluster.yml --tags postgresql_users`

**Comment by blackyzero** [comment]
I get following

```
TASK [Include OS-specific variables] ********************************************************
ok: [192.168.169.151]
ok: [192.168.169.152]

TASK [postgresql-users : Make sure the PostgreSQL users are present] ************************
failed: [192.168.169.151] (item=postgres) => {"ansible_loop_var": "item", "changed": false, "item": {"flags": "SUPERUSER", "name": "postgres", "password": "p0stgres_pss!"}, "msg": "unable to connect to database: connection to server on socket \"/var/run/postgresql/.s.PGSQL.5432\" failed: FATAL:  role \"postgres\" does not exist\n"}
...ignoring

PLAY RECAP **********************************************************************************
192.168.169.151            : ok=9    changed=0    unreachable=0    failed=0    skipped=6    rescued=0    ignored=1
192.168.169.152            : ok=8    changed=0    unreachable=0    failed=0    skipped=7    rescued=0    ignored=0
192.168.169.153            : ok=2    changed=0    unreachable=0    failed=0    skipped=3    rescued=0    ignored=0


```

**Comment by vitabaks** [comment]
ok I'll fix it

**Comment by vitabaks** [comment]
Fixed 83a2f3ec2de5e5cdca80a6ca1729fa375e47bb59

please download the playbook again and test it

**Comment by blackyzero** [comment]
i am still getting following error with new deployment.

```
# sudo -u postgres psql
psql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "postgres" does not exist
```

**Comment by vitabaks** [comment]
> error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432"

we no longer use socket connection for this task
https://github.com/vitabaks/postgresql_cluster/blob/master/roles/postgresql-users/tasks/main.yml#L11

TCP only - 127.0.0.1

please download the playbook again

**Comment by vitabaks** [comment]
Did you specify the user in the variables ?

```yaml
postgresql_users:
  - {name: "postgres", password: "postgres-pass", flags: "SUPERUSER"}
```

**Comment by blackyzero** [comment]
yes, i already added this line before running deployment.

**Comment by vitabaks** [comment]
ok i will try to reproduce and fix it a bit later

**Comment by blackyzero** [comment]
thanks @vitabaks 
FYI, i also retry the deployment, and also getting same.

```
TASK [patroni : Disable "postgresql@14-main" service] ***************************************
ok: [192.168.169.152]
ok: [192.168.169.151]

TASK [patroni : Add PATRONICTL_CONFIG_FILE environment variable into /etc/environment] ******
ok: [192.168.169.152]
ok: [192.168.169.151]

TASK [postgresql-users : Make sure the PostgreSQL users are present] ************************
failed: [192.168.169.151] (item=postgres) => {"ansible_loop_var": "item", "changed": false, "item": {"flags": "SUPERUSER", "name": "postgres", "password": "p0stgres_p@sS121!"}, "msg": "unable to connect to database: connection to server on socket \"/var/run/postgresql/.s.PGSQL.5432\" failed: FATAL:  role \"postgres\" does not exist\n"}
...ignoring

TASK [pgbouncer/userlist : Get users and password md5 from pg_shadow] ***********************
ok: [192.168.169.151]

TASK [pgbouncer/userlist : Generate /etc/pgbouncer/userlist.txt] ****************************
changed: [192.168.169.152]
changed: [192.168.169.151]

RUNNING HANDLER [pgbouncer/userlist : Reload pgbouncer service] *****************************
changed: [192.168.169.152]
changed: [192.168.169.151]

TASK [deploy-finish : Check postgresql cluster health] **************************************
ok: [192.168.169.151]

TASK [deploy-finish : PostgreSQL Cluster health] ********************************************
ok: [192.168.169.151] => {
    "patronictl_result.stdout_lines": [
        "+---------+-----------------+--------------+---------+----+-----------+",
        "| Member  | Host            | Role         | State   | TL | Lag in MB |",
        "+ Cluster: postgres-cluster (7090936482763241882) ---+----+-----------+",
        "| pgsql01 | 192.168.169.151 | Leader       | running |  1 |           |",
        "| pgsql03 | 192.168.169.152 | Sync Standby | running |  1 |         0 |",
        "+---------+-----------------+--------------+---------+----+-----------+"
    ]
}

TASK [deploy-finish : Get postgresql database list] *****************************************
ok: [192.168.169.151]

TASK [deploy-finish : PostgreSQL list of databases] *****************************************
ok: [192.168.169.151] => {
    "dbs_result.stdout_lines": [
        "   name   |    owner    | encoding |   collate   |    ctype    |  size   | tablespace ",
        "----------+-------------+----------+-------------+-------------+---------+------------",
        " postgres | patroni_usr | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 8553 kB | pg_default",
        "(1 row)"
    ]
}

TASK [deploy-finish : Create list of nodes] *************************************************
ok: [192.168.169.151]

TASK [deploy-finish : PostgreSQL Cluster connection info] ***********************************
ok: [192.168.169.151] => {
    "msg": [
        "+------------------------------------------------+",
        "address 192.168.169.151,192.168.169.152",
        "port 6432 (pgbouncer)",
        "+------------------------------------------------+"
    ]
}

PLAY RECAP **********************************************************************************
192.168.169.151            : ok=105  changed=7    unreachable=0    failed=0    skipped=348  rescued=0    ignored=1
192.168.169.152            : ok=93   changed=6    unreachable=0    failed=0    skipped=336  rescued=0    ignored=0
192.168.169.153            : ok=26   changed=1    unreachable=0    failed=0    skipped=113  rescued=0    ignored=0
localhost                  : ok=0    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0

root@ansible:/opt/services/ansible/postgresql_cluster#
```

```
root@pgsql01:~# sudo -u postgres psql
could not change directory to "/root": Permission denied
psql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "postgres" does not exist
```

**Comment by blackyzero** [comment]
Another thing, it seems the default folder is not pickup properly after changing OS "Debian" variable 

```
# PostgreSQL variables
postgresql_cluster_name: "main"
postgresql_data_dir: "/opt/data/postgresql/{{ postgresql_version }}/{{ postgresql_cluster_name }}"  # You can specify custom data dir path
postgresql_wal_dir: ""  # custom WAL dir path (symlink will be created) [optional]
postgresql_conf_dir: "/etc/postgresql/{{ postgresql_version }}/{{ postgresql_cluster_name }}"
postgresql_bin_dir: "/usr/lib/postgresql/{{ postgresql_version }}/bin"
postgresql_log_dir: "/var/log/postgresql"
postgresql_unix_socket_dir: "/var/run/postgresql"
postgresql_home_dir: "/opt/data/postgresql"

# stats_temp_directory (mount the statistics directory in tmpfs)
postgresql_stats_temp_directory_path: "/opt/data/pgsql_stats_tmp"  # or 'none'
postgresql_stats_temp_directory_size: "1024m"

```

Testing to redeploy using default username "postgres"
```
patroni_superuser_username: "postgres"
```

And getting following error:
```
Apr 27 03:24:46 pgsql01 patroni[1504265]: 2022-04-27 03:24:46,472 INFO: Selected new etcd server http://192.168.169.152:2379
Apr 27 03:24:46 pgsql01 patroni[1504265]: 2022-04-27 03:24:46,480 INFO: No PostgreSQL configuration items changed, nothing to reload.
Apr 27 03:24:46 pgsql01 patroni[1504265]: 2022-04-27 03:24:46,519 INFO: Lock owner: None; I am pgsql01
Apr 27 03:24:46 pgsql01 patroni[1504265]: 2022-04-27 03:24:46,551 INFO: trying to bootstrap a new cluster
Apr 27 03:24:46 pgsql01 patroni[1504305]: The files belonging to this database system will be owned by user "postgres".
Apr 27 03:24:46 pgsql01 patroni[1504305]: This user must also own the server process.
Apr 27 03:24:46 pgsql01 patroni[1504305]: The database cluster will be initialized with locale "en_US.UTF-8".
Apr 27 03:24:46 pgsql01 patroni[1504305]: The default text search configuration will be set to "english".
Apr 27 03:24:46 pgsql01 patroni[1504305]: Data page checksums are enabled.
Apr 27 03:24:46 pgsql01 patroni[1504305]: fixing permissions on existing directory /opt/data/postgresql/14/main ... ok
Apr 27 03:24:46 pgsql01 patroni[1504305]: creating subdirectories ... ok
Apr 27 03:24:46 pgsql01 patroni[1504305]: selecting dynamic shared memory implementation ... posix
Apr 27 03:24:46 pgsql01 patroni[1504305]: selecting default max_connections ... 100
Apr 27 03:24:47 pgsql01 patroni[1504305]: selecting default shared_buffers ... 128MB
Apr 27 03:24:47 pgsql01 patroni[1504305]: selecting default time zone ... UTC
Apr 27 03:24:47 pgsql01 patroni[1504305]: creating configuration files ... ok
Apr 27 03:24:47 pgsql01 patroni[1504305]: running bootstrap script ... ok
Apr 27 03:24:48 pgsql01 patroni[1504305]: performing post-bootstrap initialization ... ok
Apr 27 03:24:49 pgsql01 patroni[1504305]: syncing data to disk ... ok
Apr 27 03:24:49 pgsql01 patroni[1504305]: initdb: warning: enabling "trust" authentication for local connections
Apr 27 03:24:49 pgsql01 patroni[1504305]: You can change this by editing pg_hba.conf or using the option -A, or
Apr 27 03:24:49 pgsql01 patroni[1504305]: --auth-local and --auth-host, the next time you run initdb.
Apr 27 03:24:49 pgsql01 patroni[1504305]: Success. You can now start the database server using:
Apr 27 03:24:49 pgsql01 patroni[1504305]:     /usr/lib/postgresql/14/bin/pg_ctl -D /opt/data/postgresql/14/main -l logfile start
Apr 27 03:24:49 pgsql01 patroni[1504344]: 2022-04-27 03:24:49.500 GMT [1504344] LOG:  skipping missing configuration file "/var/lib/postgresql/14/main/postgresql.auto.conf"
Apr 27 03:24:49 pgsql01 patroni[1504344]: 2022-04-26 23:24:49 EDT [1504344-2]  FATAL:  data directory "/var/lib/postgresql/14/main" does not exist
Apr 27 03:24:49 pgsql01 patroni[1504265]: 2022-04-27 03:24:49,518 INFO: postmaster pid=1504344
Apr 27 03:24:49 pgsql01 patroni[1504345]: /var/run/postgresql:5432 - no response
Apr 27 03:24:49 pgsql01 patroni[1504265]: 2022-04-27 03:24:49,540 INFO: removing initialize key after failed attempt to bootstrap the cluster
Apr 27 03:24:49 pgsql01 patroni[1504265]: 2022-04-27 03:24:49,551 INFO: renaming data directory to /opt/data/postgresql/14/main_2022-04-27-03-24-49
Apr 27 03:24:49 pgsql01 patroni[1504265]: Traceback (most recent call last):
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/bin/patroni", line 8, in <module>
Apr 27 03:24:49 pgsql01 patroni[1504265]:     sys.exit(main())
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/__main__.py", line 143, in main
Apr 27 03:24:49 pgsql01 patroni[1504265]:     return patroni_main()
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/__main__.py", line 135, in patroni_main
Apr 27 03:24:49 pgsql01 patroni[1504265]:     abstract_main(Patroni, schema)
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/daemon.py", line 100, in abstract_main
Apr 27 03:24:49 pgsql01 patroni[1504265]:     controller.run()
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/__main__.py", line 105, in run
Apr 27 03:24:49 pgsql01 patroni[1504265]:     super(Patroni, self).run()
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/daemon.py", line 59, in run
Apr 27 03:24:49 pgsql01 patroni[1504265]:     self._run_cycle()
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/__main__.py", line 108, in _run_cycle
Apr 27 03:24:49 pgsql01 patroni[1504265]:     logger.info(self.ha.run_cycle())
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/ha.py", line 1503, in run_cycle
Apr 27 03:24:49 pgsql01 patroni[1504265]:     info = self._run_cycle()
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/ha.py", line 1377, in _run_cycle
Apr 27 03:24:49 pgsql01 patroni[1504265]:     return self.post_bootstrap()
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/ha.py", line 1269, in post_bootstrap
Apr 27 03:24:49 pgsql01 patroni[1504265]:     self.cancel_initialization()
Apr 27 03:24:49 pgsql01 patroni[1504265]:   File "/usr/local/lib/python3.9/dist-packages/patroni/ha.py", line 1262, in cancel_initialization
Apr 27 03:24:49 pgsql01 patroni[1504265]:     raise PatroniFatalException('Failed to bootstrap cluster')
Apr 27 03:24:49 pgsql01 patroni[1504265]: patroni.exceptions.PatroniFatalException: 'Failed to bootstrap cluster'
Apr 27 03:24:49 pgsql01 systemd[1]: patroni.service: Main process exited, code=exited, status=1/FAILURE
```

Please correct to support custom folder other than "/var/lib/postgresql".
Thank you.

**Comment by vitabaks** [comment]
create separate issues for unrelated issues. This does not apply to "Unable to access sql terminal"

**Comment by blackyzero** [comment]
You're right. i have just created another issue regarding this.
Thanks

**Comment by vitabaks** [comment]
Hello @blackyzero !

Fixed: 0752de87b26462edc5d19a09437880b0705abbff

```
TASK [postgresql-users : Make sure the PostgreSQL users are present] ********************************************************************************************************************************************
changed: [172.31.1.188] => (item=postgres)
changed: [172.31.1.188] => (item=mydb-user)
```

```
ubuntu@pgnode01:~/postgresql_cluster$ sudo -u postgres psql
psql (14.2 (Ubuntu 14.2-1.pgdg20.04+1+b1))
Type "help" for help.

postgres=# \du
                                        List of roles
     Role name      |                         Attributes                         | Member of 
--------------------+------------------------------------------------------------+-----------
 mydb-user          |                                                            | {}
 patroni_replicator | Replication                                                | {}
 patroni_usr        | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
 postgres           | Superuser                                                  | {}
```

**Comment by blackyzero** [comment]
great !
thanks @vitabaks
